### PR TITLE
Impose solution for k=0 modes

### LINKED ISSUE #2: OneDWaveEquation can't deal with the (k, l)=(0, 0) component of solution
I think one needs to be a bit careful. I'm wondering whether FourierFlows.jl handles functions with non-zero mean. Notice the animation of the [two gaussians](https://fourierflows.github.io/OscillatoryFlows.jl/dev/generated/two_gaussians/): for t>0 you see that most of the amplitude away from the gaussian is not at ξ=0 but slightly below zero. I think this has something to do with inverting from Fourier space...

Or it may be related to this line
https://github.com/FourierFlows/OscillatoryFlows.jl/blob/08f05b24d6f50014374cccd981646a1e9c7326a9/src/OneDWaveEquation.jl#L126
Why is this line there?

### PR DESCRIPTION
**Pull request by navidcy**
closes #2

### CODE REVIEW

File: src/OneDWaveEquation.jl [hunk 1]
```diff
@@ -85,11 +85,12 @@ function WaveEquation(c, grid)
...
-     ξ :: A
-     u :: A
-    ξh :: B
-    uh :: B
+struct Vars{A, B, A2} <: AbstractVars
+          ξ :: A
+          u :: A
+         ξh :: B
+         uh :: B
+  ξ0u0_mean :: A2
```
Comment (glwagner) [inline src/OneDWaveEquation.jl]: Is it necessary to store these in one variable?

Can make the `struct` mutable with variables `Ξ` and `U`?

Or, `mean_ξ` and `mean_u`.

File: src/OneDWaveEquation.jl [hunk 2]
```diff
@@ -100,7 +101,8 @@ Returns the vars for a one-dimensional wave equation problem problem on `grid`.
 function Vars(::Dev, grid::AbstractGrid{T}) where {Dev, T}
     @devzeros Dev T grid.nx ξ u
     @devzeros Dev Complex{T} grid.nkr ξh uh
-    return Vars(ξ, u, ξh, uh)
+    @devzeros Dev Complex{T} 2 ξ0u0_mean
```
Comment (glwagner) [inline src/OneDWaveEquation.jl]: Why Complex?

**Comment by navidcy** [comment]
Have a look at the [docs example](https://fourierflows.github.io/OscillatoryFlows.jl/previews/PR3/generated/two_gaussians/)

File: src/OneDWaveEquation.jl (hunk 2 above)
Comment (navidcy) [reply src/OneDWaveEquation.jl]: converted to T

File: src/OneDWaveEquation.jl (hunk 1 above)
Comment (navidcy) [reply src/OneDWaveEquation.jl]: done

File: src/OneDWaveEquation.jl [hunk 3]
```diff
@@ -85,11 +85,17 @@ function WaveEquation(c, grid)
...
-     ξ :: A
-     u :: A
-    ξh :: B
-    uh :: B
+struct Vars{A, B, S} <: AbstractVars
+                    ξ :: A
+                    u :: A
+                   ξh :: B
+                   uh :: B
+  integral_contraints :: S
```
Comment (glwagner) [inline src/OneDWaveEquation.jl]: typo: should be `constraints`

File: src/OneDWaveEquation.jl [hunk 4]
```diff
@@ -85,11 +85,17 @@ function WaveEquation(c, grid)
...
+end
+
+mutable struct IntegralConstraints{T} <: AbstractVars
```
Comment (glwagner) [inline src/OneDWaveEquation.jl:96]: so nice :-D

File: src/OneDWaveEquation.jl (hunk 3 above)
Comment (navidcy) [reply src/OneDWaveEquation.jl]: 022214f

**Pull request merged**